Main.elm

```diff
@@ -108,23 +127,77 @@ update message model =
             { model | lock = True, content = "", passphrase = "", error = "Wrong passphrase" } ! []
 
         Lock ->
-            { model | lock = True, content = "", passphrase = "" } ! [ setData model.content ]
+            { model | lock = True, content = "", passphrase = "" } ! [ encryptData { content = model.content, passphrase = model.passphrase } ]
 
-        DataSaved _ ->
-            { model | modified = False } ! []
-
-        DataNotSaved error ->
-            { model | error = (Debug.log "" error) } ! []
+        DataSaved result ->
+            let
+                _ =
+                    Debug.log "kinto result" result
+            in
+                { model | modified = False } ! []
 
         ToggleReveal ->
             { model | reveal = not model.reveal } ! []
 
         TimeOut debounceCount ->
             if debounceCount == model.debounceCount then
-                { model | debounceCount = 0 } ! [ setData model.content ]
+                { model | debounceCount = 0 } ! [ encryptData { content = model.content, passphrase = model.passphrase } ]
             else
                 model ! []
 
+        DataEncrypted encrypted ->
+            { model | encryptedData = Debug.log "encrypted data from js" <| Just encrypted } ! [ saveData model.email model.passphrase encrypted ]
+
+        DataNotEncrypted error ->
+            { model | error = (Debug.log "" error) } ! []
+
+
+
+-- Kinto related
+
+
+recordResource : Kinto.Resource String
+recordResource =
+    Kinto.recordResource
+        "default"
+        "hoverpad"
+        (Decode.field "content" Decode.string)
+
+
+decodeContent : Decode.Decoder String
+decodeContent =
+    (Decode.field "content" Decode.string)
+
+
+encodeContent : String -> Encode.Value
+encodeContent content =
+    Encode.object [ ( "content", Encode.string content ) ]
+
+
+saveData : String -> String -> String -> Cmd Msg
+saveData email passphrase content =
+    let
+        data =
+            encodeContent content
+
+        client =
+            Kinto.client kintoServer (Kinto.Basic email passphrase)
```

Comment:
We should not use the passphrase over BasicAuth because the passphrase cannot be send at anytime to the server. We can use an hashed passphrase here.